internal/noop_registry.go

```diff
@@ -0,0 +1,85 @@
+package internal
+
+type NoOpRegistry struct{}
```

Comment:
Do we see any compiler issues if we leave this unexported (i.e. `noOpRegistry`)?

Reply:
No issues.